dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostFactoryImpl.java

```diff
@@ -0,0 +1,790 @@
+    public HostFactoryImpl() {
+        this.siteCache = CacheLocator.getHostCache();
+    }
+
+    /**
+     * Lazy initialization of the Contentlet Factory service. This helps prevent startup issues when several factories
+     * or APIs are initialized during the initialization phase of the Host Factory.
+     *
+     * @return An instance of the {@link ContentletFactory} service.
+     */
+    protected ContentletFactory getContentletFactory() {
+        if (null == this.contentFactory) {
+            this.contentFactory = FactoryLocator.getContentletFactory();
+        }
+        return this.contentFactory;
+    }
+
+    /**
+     * Lazy initialization of the Contentlet API service. This helps prevent startup issues when several factories or
+     * APIs are initialized during the initialization phase of the Host Factory.
+     *
+     * @return An instance of the {@link ContentletAPI} service.
+     */
+    protected ContentletAPI getContentletAPI() {
+        if (null == this.contentletAPI) {
+            this.contentletAPI = APILocator.getContentletAPI();
+        }
+        return this.contentletAPI;
+    }
+
+    @Override
+    public Host bySiteName(final String siteName) {
+        Host site = siteCache.get(siteName);
+        if (null == site || !UtilMethods.isSet(site.getIdentifier())) {
+            final DotConnect dc = new DotConnect();
+            final StringBuilder sqlQuery = new StringBuilder().append(SELECT_SITE_INODE)
+                    .append(WHERE);
+            sqlQuery.append(getSiteNameColumn(SITE_NAME_EQUALS.toString()));
+            dc.setSQL(sqlQuery.toString());
+            dc.addParam(siteName.toLowerCase());
+            try {
+                final List<Map<String, String>> dbResults = dc.loadResults();
+                if (dbResults.isEmpty()) {
+                    return null;
+                }
+                final String siteInode = dbResults.get(0).get("inode");
+                if (dbResults.size() > 1) {
+                    // This situation should NOT happen at all
+                    final StringBuilder warningMsg = new StringBuilder().append("ERROR: ").append(dbResults.size())
+                            .append(" Sites have the same name '").append(siteName).append("':\n");
+                    for (final Map<String, String> siteInfo : dbResults) {
+                        warningMsg.append("-> Inode = ").append(siteInfo.get("inode")).append("\n");
+                    }
+                    warningMsg.append("Defaulting to Site '").append(siteInode).append("'");
+                    Logger.fatal(this, warningMsg.toString());
+                }
+                final Contentlet siteAsContentlet = this.contentFactory.find(siteInode);
+                site = new Host(siteAsContentlet);
+                this.siteCache.add(site);
+            } catch (final Exception e) {
+                throw new DotRuntimeException(String.format("An error occurred when retrieving Site by name '%s': %s",
```

Comment:
log me with input params

Reply:
The only input parameter: `siteName` is already present in the error msg.